crates/cxx-qt-gen/src/naming/name.rs

```diff
@@ -136,34 +136,56 @@ impl Name {
         }
 
         // Find if there is a cxx_name mapping (for C++ generation)
-        let mut cxx_name = attribute_find_path(attrs, &["cxx_name"])
+        let cxx_name = attribute_find_path(attrs, &["cxx_name"])
             .map(|index| -> Result<_> {
                 expr_to_string(&attrs[index].meta.require_name_value()?.value)
             })
             .transpose()?;
 
         // Find if there is a rust_name mapping
-        let rust_ident = if let Some(index) = attribute_find_path(attrs, &["rust_name"]) {
+        let rust_name = attribute_find_path(attrs, &["rust_name"])
+            .map(|index| -> Result<_> {
+                Ok(format_ident!(
+                    "{}",
+                    expr_to_string(&attrs[index].meta.require_name_value()?.value)?,
+                    span = attrs[index].span()
+                ))
+            })
+            .transpose()?;
+
+        Ok(Self {
+            rust: ident.clone(),
+            cxx: None,
+            namespace,
+            module: module.cloned().map(Path::from),
+        }
+        .with_options(cxx_name, rust_name, false))
+    }
+
+    /// Applies naming options to an existing name, applying logic about what should cause renaming
+    pub fn with_options(self, cxx: Option<String>, rust: Option<Ident>, auto_camel: bool) -> Self {
```

Comment:
generally i prefer to use enums rather than bools in method args, otherwise you can end up with `method(true, false, true)` which has no meaning compared to `method(AutoCamel::Enabled, ...)` etc.

I do however note that this auto_camel arg might disappear in the other change? But then it may also come back if we add `auto_camel` as an option to the bridge :-)

Reply:
Done